src/panels/config/core/ha-config-url-form.ts

```diff
@@ -43,46 +85,143 @@ class ConfigUrlForm extends LitElement {
               `
             : ""}
           ${this._error ? html`<div class="error">${this._error}</div>` : ""}
-          <div class="row">
-            <div class="flex">
-              ${this.hass.localize(
-                "ui.panel.config.core.section.core.core_config.external_url"
-              )}
-            </div>
 
-            <paper-input
-              class="flex"
-              .label=${this.hass.localize(
-                "ui.panel.config.core.section.core.core_config.external_url"
-              )}
-              name="external_url"
-              type="url"
-              .disabled=${disabled}
-              .value=${this._externalUrlValue}
-              @value-changed=${this._handleChange}
-            >
-            </paper-input>
+          <div class="description">
+            ${this.hass.localize("ui.panel.config.url.description")}
           </div>
 
+          ${hasCloud
+            ? html`
+                <div class="row">
+                  <div class="flex">
+                    ${this.hass.localize(
+                      "ui.panel.config.url.external_url_label"
+                    )}
+                  </div>
+                  <ha-formfield
+                    .label=${this.hass.localize(
+                      "ui.panel.config.url.external_use_ha_cloud"
+                    )}
+                  >
+                    <ha-switch
+                      .disabled=${disabled}
+                      .checked=${hasCloud && externalUrl === null}
```

Comment:
```suggestion
                      .checked=${externalUrl === null}
```
You only get here if `hasCloud` is true